engine/src/flutter/lib/web_ui/dev/build.dart

```diff
@@ -107,13 +115,23 @@ class BuildCommand extends Command<bool> with ArgUtils<bool> {
 
     if (isWatchMode) {
       print('Initial build done!');
-      print('Watching directory: ${libPath.relativeToCwd}/');
-      await PipelineWatcher(
-        dir: libPath.absolute,
-        pipeline: buildPipeline,
-        // Ignore font files that are copied whenever tests run.
-        ignore: (WatchEvent event) => event.path.endsWith('.ttf'),
-      ).start();
+      final Iterable<Future<void>> watchers = watchDirs
+          .map((dir) {
+            final FilePath watchPath = FilePath.fromWebUi(dir);
+            print('Watching directory: ${watchPath.relativeToCwd}/');
+            return watchPath;
+          })
+          .map(
+            (FilePath watchPath) =>
+                PipelineWatcher(
+                  dir: watchPath.absolute,
+                  pipeline: buildPipeline,
+                  // Ignore font files that are copied whenever tests run.
+                  ignore: (WatchEvent event) => event.path.endsWith('.ttf'),
+                ).start(),
+          );
+
+      await Future.wait(watchers);
```

Comment:
Can we make `PipelineWatcher` take a `List<String> dirs` and create multiple `DirectoryWatcher` instances?

Reply:
updated, i add a new option in PipelineWatcher support listen dirs